feat: allow Discover command to sort by history or alphabetically

zukuo opened the pull request:
### Summary

This PR adds an extra optional parameter to the `:NeovimProjectDiscover` command that allows you to change the ordering of the projects. Namely, two new parameters:
- `alphabetical` - default view (doesn't need to be passed in)
- `history` - prioritises most recent projects first, then all other discovered projects are added in after (alphabetically)

**Example:**
```
:NeovimProjectDiscover history
```

### Why

Wanted a way to be able to see all the projects from my directories, but still have the most recent projects show up first - mainly to have one command that could easily be used.

coffebar commented:
Thank you for your effort! 
 
I checked that it works, but I disagree with the statement that the default sorting is alphabetical. 

https://github.com/coffebar/neovim-project/blob/a37f702103acda57735b06bd9c1a056fb73f011f/README.md?plain=1#L200

I suggest you to add actually alphabetical sorting as a default, or change naming of the parameter value to something like "default" (pay attention to readme and doc/txt be consistent with the functionality)

zukuo commented:
Ah yes you're absolutely right! I'll go ahead and fix that

zukuo commented:
I've gone ahead and added a default param that just uses the config pattern order, as well as some alphabetical sorting for projects (either by their path or name). Also, updated the readme and docs.

Let me know if there is anything else you think I need to change
